tests/accelerators/test_ipu.py

```diff
@@ -581,7 +582,7 @@ def test_accelerator_ipu_with_ipus_priority():
     with pytest.warns(UserWarning, match="The flag `devices=1` will be ignored,"):
         trainer = Trainer(accelerator="ipu", devices=1, ipus=ipus)
 
-    assert trainer.ipus == ipus
+    assert trainer.num_devices == ipus
```

Comment:
This might sound repetitive, and I'm curious - whenever we are asserting `trainer.num_devices == <expected_ipus>` - should we also check if the accelerator was `IPUAccelerator` (in the tests)?

```suggestion
    assert isinstance(trainer.accelerator, IPUAccelerator)
    assert trainer.num_devices == ipus
```